app/src/main/java/com/example/taskyy/ui/viewmodels/LoginViewModel.kt

```diff
@@ -33,8 +33,18 @@ class LoginViewModel @Inject constructor(
 
     init {
         viewModelScope.launch {
-            if (authRepository.validateToken()) {
-                eventChannel.send(LoginEvent.LoginSuccess(state.email))
+            when (authRepository.validateToken()) {
+                is Result.Error -> {
+                }
+
+                is Result.Success -> {
+                    if (userPreferences.getUserFullName("name") != "" && userPreferences.getUserEmail(
+                            "email"
```

Comment:
I think I commented on this before, but determining which exact string keys to pass here shouldn't be the business of the class that uses these preferences. You can just keep these internally in the `UserPreferencesImpl` and not pass a key at all. In the end, if I want to get an email, why would I need to pass a key that matches the email?